include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,427 @@
+                  std::string const & desc,
+                  option_spec const & spec) 
+    {
+        add_option(value, 
+                   short_id,
+                   long_id,
+                   desc,
+                   spec,
+                   default_validator<bool> {});
+    }
+
+    /*!\brief Adds calls for appending 'clielement' and 'ITEM' nodes to the CTD DOM tree to be evaluated at parse time.
+     *
+     * \copydetails argument_parser::add_positional_option
+     */
+    template<typename option_type, typename validator_type>
+    void add_positional_option(option_type & value,
+                               std::string const & desc,
+                               validator_type && validator) 
+    {
+        unsigned argument_id = args_counter;
+
+        clielement_argument_callbacks.push_back([this, argument_id] (std::string app_name) 
+	{
+            append_ctd_clielement_node("",
+                                       build_reference_name(app_name,
+                                                            argument_id).data());
+        });
+
+        // Register ITEM callback.
+        item_argument_callbacks.push_back([this, value, desc, validator, argument_id] () 
+	{
+            append_ctd_item_node(build_argument_name(argument_id),
+                                 get_type_as_ctd_string(value,
+                                                        validator),
+                                 desc,
+                                 "",
+                                 "*.*",
+                                 "true",
+                                 "false",
+                                 "");
+        });
+	
+	// Increment argument counter
+        args_counter++;
+    }
+
+    /*!\brief Builds the CTD document DOM tree and prints it to standard output.
+     *
+     * \param[in] meta The meta information that are needed for building the DOM tree.
+     */
+    void parse(argument_parser_meta_data const & meta) 
+    {
+        // Build the CTD document line by line inside the string stream.
+        append_ctd_declaration_node();
+        append_ctd_tool_node(meta);
+        
+        // Print the CTD file on the standard output stream.
+        std::cout << ctd_stream.str();
+
+        std::exit(EXIT_SUCCESS);
+    }
+
+    // functions are not needed for command line parsing but are part of the format interface.
+    //!\cond
+    void add_section(std::string const &) {}
+    void add_subsection(std::string const &) {}
+    void add_line(std::string const &, bool) {}
+    void add_list_item(std::string const &, std::string const &) {}
+    //!\endcond
+
+private:
+
+    /*!\brief Build the identifier for the i-th positional option.
+     *
+     * \param[in] argument_numeric_id The numeric identifier of the argument being described.
+     * \return The string representing the i-th positional option identifier.
+     */
+    std::string
+    build_argument_name(unsigned argument_numeric_id)
```

Comment:
```suggestion
    std::string build_argument_name(unsigned argument_numeric_id)
```

and other places

Reply:
You can consolidate build_reference_name and use constexpr if blocks inside to do the right thing